### DISCUSSION TITLE
Message files per package in monorepo.

### DISCUSSION THREAD

**Discussion by zZHorizonZz** [amannn/next-intl#1603] [Q&A]
I'm using next-intl in company application. I'm currently responsible for bootstrapping the app and preparing it for future development and for other developers. We have decided on a project structure where we have a sort of "application" per package (in this context, an "application" is not a standalone thing but rather a React component that contains code and logic for a specific feature - for example, we currently have "tasks" and "laboratory" packages tasks is basically just done call to API to get data and then it's rendered in the table).

So far, we are using messages that we have defined in the Next.js application, and it works. However, from a future perspective, this would make it awkward to work with if multiple developers are working on their own packages but then they would need to edit messages in a completely different package. 

Is there a way to have message files per package?

**Comment by amannn** [comment]
This should definitely be possible, you can [merge messages](https://next-intl.dev/docs/usage/configuration#messages-split-files) at the call site of your app (which should depend on all packages in your monorepo I suppose). Have you tried that?

Closely related: https://github.com/amannn/next-intl/discussions/1224 & https://github.com/amannn/next-intl/discussions/357
